instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshStateTest.kt

```diff
@@ -0,0 +1,547 @@
+            )
+        )
+
+        createMapboxNavigation(createRouteRefreshOptionsWithInvalidInterval(10_000))
+        mapboxNavigation.registerRouteRefreshStateObserver(observer)
+        mapboxNavigation.startTripSession()
+        val requestedRoutes = requestRoutes()
+        mapboxNavigation.setNavigationRoutes(requestedRoutes)
+
+        waitForRefresh()
+
+        assertEquals(
+            listOf(
+                RouteRefreshExtra.REFRESH_STATE_STARTED,
+                RouteRefreshExtra.REFRESH_STATE_FINISHED_SUCCESS
+            ),
+            observer.getStatesSnapshot()
+        )
+    }
+
+    @Test
+    fun successfulFromSecondAttemptRefreshTest() = sdkTest {
+        setupMockRequestHandlers(
+            coordinates,
+            R.raw.route_response_single_route_refresh,
+            NthAttemptHandler(
+                createRefreshHandler(
+                    R.raw.route_response_route_refresh_annotations,
+                    "route_response_single_route_refresh"
+                ),
+                1
+            )
+        )
+
+        createMapboxNavigation(createRouteRefreshOptionsWithInvalidInterval(5_000))
+        mapboxNavigation.registerRouteRefreshStateObserver(observer)
+        mapboxNavigation.startTripSession()
+        val requestedRoutes = requestRoutes()
+        mapboxNavigation.setNavigationRoutes(requestedRoutes)
+
+        waitForRefresh()
+
+        assertEquals(
+            listOf(
+                RouteRefreshExtra.REFRESH_STATE_STARTED,
+                RouteRefreshExtra.REFRESH_STATE_FINISHED_SUCCESS
+            ),
+            observer.getStatesSnapshot()
+        )
+    }
+
+    @Test
+    fun threeFailedAttemptsThenSuccessfulRefreshTest() = sdkTest {
+        setupMockRequestHandlers(
+            coordinates,
+            R.raw.route_response_single_route_refresh_without_traffic,
+            NthAttemptHandler(
+                createRefreshHandler(
+                    R.raw.route_response_route_refresh_annotations_without_traffic,
+                    "route_response_single_route_refresh_without_traffic"
+                ),
+                3
+            )
+        )
+
+        createMapboxNavigation(createRouteRefreshOptionsWithInvalidInterval(4_000))
+        mapboxNavigation.registerRouteRefreshStateObserver(observer)
+        mapboxNavigation.startTripSession()
+        val requestedRoutes = requestRoutes()
+        mapboxNavigation.setNavigationRoutes(requestedRoutes)
+
+        waitForRefresh()
+
+        assertEquals(
+            listOf(
+                RouteRefreshExtra.REFRESH_STATE_STARTED,
+                RouteRefreshExtra.REFRESH_STATE_FINISHED_FAILED,
+                RouteRefreshExtra.REFRESH_STATE_STARTED,
+                RouteRefreshExtra.REFRESH_STATE_FINISHED_SUCCESS
+            ),
+            observer.getStatesSnapshot()
+        )
+    }
+
+    @Test
+    fun routeRefreshIsNotCancelledOnDestroyTest() = sdkTest {
```

Comment:
It's not about cancelling the job, it's about the observer. I don't expect to receive CANCELLED here because:
a) it was the behaviour before this PR;
b) all observers are removed in `onDestroy` so we would depend on the line order in `onDestroy` which is not cool although manageable.

Reply:
Got it, I think that's fine. I think the test description is just misleading - we are canceling the job, we're just not triggering the observer.

Reply:
Renamed the test.